docs/ccv1_image_security_design.md

```diff
@@ -0,0 +1,345 @@
+
+3. Verify the signature of the container image according to the configuration of the policy.json file.
+
+4. Pull the container image layer by layer. If an encrypted container image layer is encrypted, decrypt it.
+
+After the image is pulled, the subsequent deployment process is consistent with that of the normal container image.
+
+### policy.json
+
+policy.json is a necessary configuration file for deploying images in the confidential container V1 design. Its location is specified in the /etc/containers/ directory of boot image rootfs. Its main functions are as follows:
+
+- Declare a whitelist of container images with registry granularity
+
+- Configure the public key to use when verifying the image signature
+
+Thus, policy.json is confidential information that must be controlled by users and cannot be read or modified by untrusted entities.
+
+If image-rs finds that there is no policy.json in the /etc/containers/ directory or If the policy.json file is empty, call the GetResource grpc service provided by the attestation-agent. After remote attestation, obtain it from the user controlled relying KBS through the encrypted channel.
+
+The format of the policy.json file follows the format and syntax defined in the [container/image policy.json](https://github.com/containers/image/blob/main/docs/containers-policy.json.5.md) document. In the V1 design of the confidential containers, a safe and reasonable policy.json file should look like the following example: 
+
+```json
+{
+    "default": [{"type": "reject"}], 
+    "transports": {
+        "docker": {
+            "docker.io/my_private_registry": [
+                {
+                    "type": "signedBy",
+                    "keyType": "GPGKeys",
+                    "keyData": "<public Key>",
+                }
+            ],
+            "registry.access.redhat.com": [
+                {
+                    "type": "signedBy",
+                    "keyType": "GPGKeys",
+                    "keyPath": "/etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release",
+                }
+            ],
+        },
+        "dir": {
+            "": [{"type": "reject"}]
+        }
+    }
+}
+```
+
+Before pulling the container image, match the policy in the following order according to the location/registry of the container image and execute the corresponding policies:
+
+- Match single specific scopes under each member in transport:
+  - For the image from the container registry that implements the Docker Registry HTTP API V2 protocol, match each scope under the "docker" member;
+  - For images from local storage, match each scope under the "dir" member;
+- Match the default scope under each member in the transport. In the above example, there is only one default scope under the "dir" member, and the policy is reject.
+- Match the global default scope, such as the "default" field in the first line of the above example. This policy must be configured to reject.
```

Comment:
The policy file provides this flexible configuration function. If the user does not want to use any container image signature verification function, it can be configured in the policy file as follows:
```
{
  "default": [
    {
      "type": "insecureAcceptAnything"
    }
  ],
}
```
If the user specifies a registry and does not need to verify the signature, just change the policy type under the corresponding registry to `insecureAcceptAnything`. 

This will cause the signature verification system to no longer take effect, and any container image without signature will be pulled and deployed. We do not recommend this method. However, if users have special needs in this regard, they can inform the rootfs producer (such as CSP) in advance, and the rootfs producer will write the above unsafe configuration to the policy file.

Reply:
Ok, I think this is fine. Keep in mind that disabling signature verification isn't necessarily non-secure if the containers are encrypted.